packages/contracts-bedrock/scripts/deploy/Deploy.s.sol

```diff
@@ -285,6 +283,30 @@ contract Deploy is Deployer {
         // Deploy Current OPChain Contracts
         deployOpChain();
 
+        // Apply modifications for non-standard configurations not supported by the OPCM deployment
+        if (cfg.useFaultProofs()) {
+            // The OPCM doesn't deploy the CANNON game (it is deployed in deployOPChain), but the tests expect it
+            // to be the respected game type.
+            vm.prank(ISuperchainConfig(mustGetAddress("SuperchainConfigProxy")).guardian());
+            IOptimismPortal2(mustGetAddress("OptimismPortalProxy")).setRespectedGameType(GameTypes.CANNON);
```

Comment:
Why do we unconditionally deploy the FaultDisputeGame and the corresponding DelayedWETH for it in `deployOPChain`, and not here? Are there tests that run against those contracts regardless of the `useFaultProofs` value?

Reply:
I agree that we should explore that optimization, and aim for `deployOpChain` to do the minimum necessary after the last `save()` call. 

> Are there tests that run against those contracts regardless of the useFaultProofs value?

At least one example is [`Initializable.t.sol`](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/test/vendor/Initializable.t.sol#L107-L122), which is difficult to refactor with the `useX()` feature flags because it would also need to modify the [`excludes` array](https://github.com/ethereum-optimism/optimism/blob/e2e-rdme/packages/contracts-bedrock/test/vendor/Initializable.t.sol#L396-L417) depending on those flags. It can be done, but I wanted to get this up.

Reply:
Got it, will defer to you on whether to make any of these potential changes in this PR, or defer them to a future PR. Feel free to resolve this accordingly

Reply:
This will be why your tests are failing.  The respected game type should be `FAST` (254).

And for the record, it is possible to run e2e tests that use fault proofs even if the portal uses the L2OO - we ran in that mode for a long time. I think we only run e2e fault proof tests with OptimismPortal2 now (useFaultProofs = true) so from the go side it should be fine to not deploy the DisputeGameFactory at all when using the L2OO.

Reply:
> Got it, will defer to you on whether to make any of these potential changes in this PR, or defer them to a future PR. Feel free to resolve this accordingly

I have started another PR to make those changes on top of this one, but I think there may be a [bug in Initializable](https://github.com/ethereum-optimism/optimism/blob/opcm/tests/deploy-opchain/packages/contracts-bedrock/scripts/Artifacts.s.sol#L229) causing it to always look at the initialized status of the implementation. Combined with the name of the contract being "PermissionedDelayedWETHProxy", which doesn't match the source contract, that test suite requires work to enable conditionally deploying this proxy.

Resolving this for now.